### DISCUSSION TITLE
Odd behavior on Spa Termostat

### DISCUSSION THREAD

**Discussion by captain324** [cvonk/OPNpool#13] [General]
I recently built a version of this project and installed it on my Pentair Easytouch (v2.0E) panel and integrated it into my Home Assistant installation. This project replaced the project from https://github.com/thehookup/PoolMCU - nice upgrade.

Most functions are working as described here and in the design document. However, the spa thermostat exhibits an odd behavior - it flips from 93F, the heat setpoint on the easytouch controller to 4F in a random pattern. The pool thermostat works as expected. Any ideas on how to chase this down would be appreciated. Thanks

![image](https://github.com/cvonk/OPNpool/assets/133161739/280f1adf-6fd3-4420-a980-911cbdd5146e)

Target temp for spa and pool

{"mode":"heat","heatsrc":"None","target_temp":4,"current_temp":90,"action":"off"} 05/29/2023 4:57:35 PM(-1.61 seconds) 
{"mode":"heat","heatsrc":"Heater","target_temp":93,"current_temp":90,"action":"idle"} 05/29/2023 4:58:05 PM(-0.44 seconds) 

The heater source also flips from "None" to "Heater: but it doesn't line up with flips in target temp

**Comment by cvonk** [comment]
Interesting.  I suggest checking the code for F to Centigrade, and back
conversions. I used to display the temperature in C. Sorry, can't test at
the moment.

Cheers
/C

Op ma 29 mei 2023 23:02 schreef captain324 ***@***.***>:
